\chapter{Results}
\label{c:result}

\section{Datasets}

%
% GSE52194 Breast cancer
% GSE52778 Airway Muscle and Asthma
%

Two RNA-Seq datasets and one DNA-Seq datasets from Gene Expression Omnibus
(GEO) were used for the demonstration of BioCloud. GSE52194
\cite{eswaran2012:transcriptomic} was a transcriptome profiling dataset of
human breast cancer. mRNA profiles of 17 breast tumor samples of three
different subtypes (TNBC, non-TNBC and HER2-positive) and normal human breast
organoids (epithelium) samples (NBS) were sequenced using Illumina HiSeq 2000
sequencer. Full sample list can be found in Table~\ref{tab:dataset-breast}.
Raw sequence reads were obtained from NCBI Sequence Read Archive (SRA), thus
their FASTQ file names were based on their SRA accession ID. In GSE52194 breast
cancer dataset, condition was defined to be the cancer subtype of the sample.

\input{tables/dataset-breast}

Another RNA-Seq dataset, GSE52778 \cite{himes2014:rnaseq}, is a transcriptome
profiling dataset of human airway smooth muscle (HASM). mRNA profiles of 4 male
white donors' HASM cells treated with four treatment conditions were sequenced
using Illumina HiSeq 2000 sequencer with Illumina TruSeq assay, four treatment
conditions being: no treatment (Untreated); treatment with Albuterol (Alb);
treatment with Dexamethasone (Dex); treatment with simultaneous Albuterol and
Dexamethasone (Alb\_Dex). Full sample list can be found in
Table~\ref{tab:dataset-airway}. Raw sequence reads were obtained from SRA thus
sample raw FASTQ files are renamed using their SRA accession ID.

\input{tables/dataset-airway}

% TODO: how deep is the sequencing depth of these WES samples?

The DNA-Seq dataset used in the demonstration was a human whole exome
sequencing done in our lab. Exome sequencing of 5 members from the same family
using Illumina HiSeq 2000 sequencer. The study aimed to find the common
variants shared in this family.


\section{Account registration}

TODO: Screenshot groups here.

\section{Data source discovery}

% checksum check

\section{Experiment design}

% condition
% simple and advanced mode

\section{Analysis submission}


\section{Job queue monitoring}

% email notification

\section{Report and result access}

\subsection{Integration with public genome browser}


\section{RNA-Seq analysis result}

\subsection{QC}

\subsection{Alignment - STAR}

\subsection{Alignment - HISAT2}

TODO: finish the dataset execution

\subsection{Cufflinks}

\subsection{featureCounts}

\subsection{DESeq2}

TODO: finish the dataset execution


\section{DNA-Seq analysis result}

\subsection{QC}

\subsection{Alignment - BWA MEM}

\subsection{Variant calling - Varscan}

\subsection{Variant calling - GATK}

TODO: finish the data execution

% There is a tree in Figure~%\ref{i:tree}.
% This is English line spacing test. You should see double spacing text.
% This is English line spacing test. You should see double spacing text.
% This is English line spacing test. You should see double spacing text.

%i:tree
%\input{figures/tree}

% There is a barchart in Figure~%\ref{i:barchart}.
% This is English line spacing test. You should see double spacing text.
% This is English line spacing test. You should see double spacing text.
% This is English line spacing test. You should see double spacing text.

%i:barchart
%\input{figures/barchart}

% Our method outperforms state-of-art systems as shown in Table~\ref{t:results}.
% This is English line spacing test. You should see double spacing text.
% This is English line spacing test. You should see double spacing text.
% This is English line spacing test. You should see double spacing text.

%t:results
% \input{tables/results}
% \input{tables/threeparttable}
% vim: set textwidth=79 spell:
