\chapter{Results}
\label{c:result}

\section{Datasets}

%
% GSE52194 Breast cancer
% GSE52778 Airway Muscle and Asthma
%

Two RNA-Seq datasets and one DNA-Seq datasets from Gene Expression Omnibus
(GEO) were used for the demonstration of BioCloud. GSE52194
\cite{eswaran2012:transcriptomic} was a transcriptome profiling dataset of
human breast cancer. mRNA profiles of 17 breast tumor samples of three
different subtypes (TNBC, non-TNBC and HER2-positive) and normal human breast
organoids (epithelium) samples (NBS) were sequenced using Illumina HiSeq 2000
sequencer. Full sample list can be found in Table~\ref{tab:dataset-breast}.
Raw sequence reads were obtained from NCBI Sequence Read Archive (SRA), thus
their FASTQ file names were based on their SRA accession ID. In GSE52194 breast
cancer dataset, condition was defined to be the cancer subtype of the sample.

\input{tables/dataset-breast}

Another RNA-Seq dataset, GSE52778 \cite{himes2014:rnaseq}, is a transcriptome
profiling dataset of human airway smooth muscle (HASM). mRNA profiles of 4 male
white donors' HASM cells treated with four treatment conditions were sequenced
using Illumina HiSeq 2000 sequencer with Illumina TruSeq assay, four treatment
conditions being: no treatment (Untreated); treatment with Albuterol (Alb);
treatment with Dexamethasone (Dex); treatment with simultaneous Albuterol and
Dexamethasone (Alb\_Dex). Full sample list can be found in
Table~\ref{tab:dataset-airway}. Raw sequence reads were obtained from SRA thus
sample raw FASTQ files are renamed using their SRA accession ID.

\input{tables/dataset-airway}

% TODO: how deep is the sequencing depth of these WES samples?

The DNA-Seq dataset used in the demonstration was a human whole exome
sequencing done in our lab. Exome sequencing of 5 members from the same family
using Illumina HiSeq 2000 sequencer. The study aimed to find the common
variants shared in this family.



\section{Account registration and user dashboard}

A new user account was registered on BioCloud using email
\texttt{demo@biocloud.liang2.io}. Screenshots of the registration process are
shown in Figure~\ref{fig:biocloud-signup}, which resemble the process on common
websites. Most of the user inputs were sanity checked and validated, as shown
in Figure~\ref{fig:biocloud-signup}(b). After user completed the form, a
verification email was sent to the registered email address so as to make sure
all future email notifications can reach the user. A fixed message is displayed
on all pages of BioCloud if the user does not complete the email verification,
as shown in Figure~\ref{fig:biocloud-signup}(c).
Figure~\ref{fig:biocloud-signup}(d) shows the content of the verification
email, where an uniquely HMAC-based verification link was given. After clicking
on the link, user completed the verification and the fixed message for
verification was gone, as shown in Figure~\ref{fig:biocloud-signup}(e).

\input{figures/biocloud_signup}

A user dashboard is available for user-related settings as shown in
Figure~\ref{fig:biocloud-dashboard}(b). User profile including user name and
authentication number (more on Section~\ref{s:report-result-access}) can be
updated here. Password can be changed via the link in the password section in
profile update, as shown in Figure~\ref{fig:biocloud-dashboard}(b). For staff
and superuser with special permission, they can access to the admin interface
of BioCloud in user dashboard via an extra tab ``Admin'', as shown in
Figure~\ref{fig:biocloud-dashboard}(d). Admin interface will be introduced in
Section~\ref{s:biocloud-admin}.

\input{figures/biocloud_dashboard}



\section{Data source discovery}

As a newly registered user, a message was always shown to hint user to attach
their data sources to BioCloud. User can find their specific data source folder
from menu Data Source \textrightarrow List, as shown in
Figure~\ref{fig:biocloud-data-source}(b). For the demo user, location of the
data source folder on BioCloud server was:

\begin{CVerbatim}[fontsize=\small]
/biocloud/data_sources/2/
\end{CVerbatim}

\vspace{-1em}\noindent
After new data sources were added, BioCloud updated its database by comparing
the data sources in record with that in current folder.  All new discovered
data sources will be listed with guess of sample name and file type, as shown
in Figure~\ref{fig:biocloud-data-source}(a). User can pass the SHA2-256
checksum value for each file computed locally. BioCloud computed the checksum
of these files and compared the value with that supplied by the user. User can
check and update the detail of their recorded data sources, as shown in
Figure~\ref{fig:biocloud-data-source}(b). BioCloud also guessed the strand for
pair-end FASTA/Q files. The information was recorded at the metadata column in
JSON format.

\input{figures/biocloud_data_source}



\section{Experiment design}

Figure~\ref{fig:biocloud-experiment-design}.

% condition
\input{figures/biocloud_experiment_design}

% simple and advanced mode
Figure~\ref{fig:biocloud-experiment-sample-condition}.

\input{figures/biocloud_experiment_sample_condition}


\section{Analysis submission}


\section{Job queue monitoring}

% email notification

\section{Report and result access}
\label{s:report-result-access}

\subsection{Integration with public genome browser}


\section{Admin}
\label{s:biocloud-admin}


\section{RNA-Seq analysis result}

\subsection{QC}

\subsection{Alignment - STAR}

\subsection{Alignment - HISAT2}

TODO: finish the dataset execution

\subsection{Cufflinks}

\subsection{featureCounts}

\subsection{DESeq2}

TODO: finish the dataset execution


\section{DNA-Seq analysis result}

\subsection{QC}

\subsection{Alignment - BWA MEM}

\subsection{Variant calling - Varscan}

\subsection{Variant calling - GATK}

TODO: finish the data execution

% There is a tree in Figure~%\ref{i:tree}.
% This is English line spacing test. You should see double spacing text.
% This is English line spacing test. You should see double spacing text.
% This is English line spacing test. You should see double spacing text.

%i:tree
%\input{figures/tree}

% There is a barchart in Figure~%\ref{i:barchart}.
% This is English line spacing test. You should see double spacing text.
% This is English line spacing test. You should see double spacing text.
% This is English line spacing test. You should see double spacing text.

%i:barchart
%\input{figures/barchart}

% Our method outperforms state-of-art systems as shown in Table~\ref{t:results}.
% This is English line spacing test. You should see double spacing text.
% This is English line spacing test. You should see double spacing text.
% This is English line spacing test. You should see double spacing text.

%t:results
% \input{tables/results}
% \input{tables/threeparttable}
% vim: set textwidth=79 spell:
